hardware/raspberrypi/bootmodes/net.md

```diff
@@ -134,6 +134,10 @@ The Raspberry Pi will only respond to ARP requests when it is in the initialsati
 
 Fixed in Raspberry Pi 3 Model B+.
 
+### DHCP request / reply / ack sequence not correctly implemented
+
+At boot time, Raspberry Pi broadcasts a DHCPDISCOVER packet.  The DHCP server then replies with a DHCPOFFER packet, the Pi then continues booting without doing a DHCPREQUEST or waiting for DHCPACK.  The can mean two separate devices can be offered the same IP address and will use it without it being properly assigned to the client.  Different DHCP servers have different behaviours in this situation, dnsmasq (depending upon settings) will hash the MAC address to determine the IP address and ping it to make sure it isn't already in use.  So the chance this happens is reduced because it requires a collision in the hash.
+
```

Comment:
No, the bootrom will always use the IP address without doing a DHCPREQUEST,  you could do the request in bootcode.bin after it has loaded this bit...  But it's sounds like a waste of time since you still won't be able to guarantee it'll work

Reply:
Ah - I obviously have no idea what parts are done by the bootrom and what parts are done by 
`bootcode.bin` :stuck_out_tongue_winking_eye: 
I assumed the bits that mention "no workarounds other than to use an SD card with just the bootcode.bin file" meant that parts of bootcode.bin (on the SD card) 'overrode' what was in the bootrom? :man_shrugging: